node/core/backing/src/lib.rs

```diff
@@ -0,0 +1,1570 @@
+			FromJob::Provisioner(msg) => AllMessages::Provisioner(msg),
+		}
+	}
+}
+
+// It looks like it's not possible to do an `impl From` given the current state of
+// the code. So this does the necessary conversion.
+fn primitive_statement_to_table(s: &SignedFullStatement) -> TableSignedStatement {
+	let statement = match s.payload() {
+		Statement::Seconded(c) => TableStatement::Candidate(c.clone()),
+		Statement::Valid(h) => TableStatement::Valid(h.clone()),
+		Statement::Invalid(h) => TableStatement::Invalid(h.clone()),
+	};
+
+	TableSignedStatement {
+		statement,
+		signature: s.signature().clone(),
+		sender: s.validator_index(),
+	}
+}
+
+// finds the first key we are capable of signing with out of the given set of validators,
+// if any.
+fn signing_key(validators: &[ValidatorId], keystore: &KeyStorePtr) -> Option<ValidatorPair> {
+	let keystore = keystore.read();
+	validators.iter()
+		.find_map(|v| {
+			keystore.key_pair::<ValidatorPair>(&v).ok()
+		})
+}
+
+impl CandidateBackingJob {
+	/// Run asynchronously.
+	async fn run(mut self) -> Result<(), Error> {
+		while let Some(msg) = self.rx_to.next().await {
+			match msg {
+				ToJob::CandidateBacking(msg) => {
+					self.process_msg(msg).await?;
+				}
+				_ => break,
+			}
+		}
+
+		Ok(())
+	}
+
+	async fn issue_candidate_invalid_message(
+		&mut self,
+		candidate: AbridgedCandidateReceipt,
+	) -> Result<(), Error> {
+		self.tx_from.send(FromJob::CandidateSelection(
+			CandidateSelectionMessage::Invalid(self.parent, candidate)
+		)).await?;
+
+		Ok(())
+	}
+
+	/// Validate the candidate that is requested to be `Second`ed and distribute validation result.
+	async fn validate_and_second(
+		&mut self,
+		candidate: AbridgedCandidateReceipt,
+		pov: PoVBlock,
+	) -> Result<ValidationResult, Error> {
+		let valid = self.request_candidate_validation(candidate.clone(), pov.clone()).await?;
+		let statement = match valid.0 {
+			ValidationResult::Valid => {
+				// make PoV available for later distribution. Send data to the availability
+				// store to keep. Sign and dispatch `valid` statement to network if we
+				// have not seconded the given candidate.
+				self.make_pov_available(pov, valid.1, valid.2).await?;
+				self.issued_validity.insert(candidate.hash());
+				Statement::Seconded(candidate)
+			}
+			ValidationResult::Invalid => {
+				let candidate_hash = candidate.hash();
+				self.issue_candidate_invalid_message(candidate).await?;
+				Statement::Invalid(candidate_hash)
+			}
+		};
+
+		if let Some(signed_statement) = self.sign_statement(statement) {
+			self.import_statement(&signed_statement).await?;
+			self.distribute_signed_statement(signed_statement).await?;
+		}
+
+		Ok(valid.0)
+	}
+
+	fn get_backed(&self) -> Vec<NewBackedCandidate> {
+		let proposed = self.table.proposed_candidates(&self.table_context);
+		let mut res = Vec::with_capacity(proposed.len());
+
+		for p in proposed.into_iter() {
+			let TableAttestedCandidate { candidate, validity_votes, .. } = p;
```

Comment:
```suggestion
		for TableAttestedCandidate { candidate, validity_votes, .. } in proposed.into_iter() {
```

Reply:
That line might be too long but I agree with it in spirit.